Week1/js-exercises/ex5-catWalk.js

```diff
@@ -9,5 +9,47 @@
  4. Call that function every 50 milliseconds.Your cat should now be moving across the screen from left to right.Hurrah!
  5. When the cat reaches the right - hand of the screen, restart them at the left hand side("0px").So they should keep walking from left to right across the screen, forever and ever.
  6. When the cat reaches the middle of the screen, replace the img with an image of a cat dancing(use this URL: https: //tenor.com/StFI.gif), keep it dancing for 5 seconds, and then replace the img with the original image and have it continue the walk.
- 
-*/
\ No newline at end of file
+
+*/
+
+const imgCat = document.getElementsByTagName('img')[0];
+imgCat.style.left = '0px';
+let speed = 1;
+
+const winScreen = window.innerWidth-imgCat.width;
+
+const catMoves = setInterval(catWalk, 5)
+
+function catWalk() {
```

Comment:
Does this work even when you have defined the function `catWalk` after using it?

Reply:
I haven't finished this exercise yet! Actually I'm stuck in this one!